src/chromo/util.py

```diff
@@ -733,3 +810,94 @@ def select_long_lived(tau=0, mm=False):
             long_lived.append(pid)
 
     return long_lived
+
+
+class redirect_std_streams:
+    """
+    A context manager for redirecting standard output (stdout)
+    and standard error (stderr) streams to files or /dev/null.
+
+    If file exists, the redirected content is appended, otherwise
+    the file is created with all its parent directories
+
+    Usage:
+    ```
+    with redirect_std_streams(stdout_file="out.txt", stderr_file="err.txt"):
+        ...
+    ```
+
+    Taken from https://stackoverflow.com/a/978264 and
+    https://stackoverflow.com/a/50691621
+    """
+
+    def __init__(self, *, stdout_file=None, stderr_file=None):
+        """
+        Initialize the redirect_std_streams context manager.
+
+        Args:
+            stdout_file (Path or str): The file to redirect stdout to.
+            If 'null', stdout will be redirected to /dev/null.
+
+            stderr_file (Path or str): The file to redirect stderr to.
+            If 'null', stderr will be redirected to /dev/null.
+        """
+        self.stdout_file = stdout_file
+        self.stderr_file = stderr_file
+        # Descriptors for stdout and stderr
+        self.stdout = 1
+        self.stderr = 2
+
+    def _file(self, file_name):
+        """
+        Create a file with the specified name and return its Path object.
+        """
+        file = Path(file_name)
+        file.parent.mkdir(parents=True, exist_ok=True)
+        return file
+
+    def __enter__(self):
+        """
+        Enter the context and redirect stdout and stderr
+        to the specified files or /dev/null.
+        """
+        if self.stdout_file:
+            if self.stdout_file == "null":
```

Comment:
The magic string "null" is bad design, and it is not clearly undocumented. Please check for equality with `os.devnull` instead. `os.devnull` is also a string, but since it supported by `os`, it is at least part of the Python standard libs.